### PR TITLE
Add command for adding/removing roles and targets and refactor cli

### LINKED ISSUE #9: Add support for handling `repositories.json`
`repositories.json` is a target file which must exist in a TUF repository in order to use TAF. So, it seems like a good idea to implement functionalities for updating this file. Some ideas that come to mind mind are:
- Add a new repository
- Remove an old repository
- Add/edit/remove url
- Add/edit/remove custom data

- [ ] GUI
- [ ] CLI

### LINKED ISSUE #312: Automate add/remove role and target repository
Create commands which would fully handle:
   - addition of new target repositories (support specification of a new role)
   - removal of a target repository
   - listing all target repositories
   - adding new roles
   - removing roles

### PR DESCRIPTION
**Pull request by renatav**
## Description (e.g. "Related to ...", etc.)

- Major refactoring of the cli commands and the `developer_tool` module, which used to contain every function and helper function needed by the cli. Moved more than a half of this module into new `api` modules, organized by their domain (targets, roles, metadata) and existing keys handling utils.
- Removed some deprecated commands, like generating repositories json
- Added commands for adding and removing roles and target repositories and for listing target repositories. There can still be more work done on the terminology and input parsing, raising nicer errors etc. but the commands should always leave the authentication repositories in a valid state, appropriately updating `repositories.json`, target files, signing and automatically committing.
- Some examples:
- `taf roles add-role law test --path test --path test2 --keystore ..\keystore-dev --keys-number 3 --threshold 2`
- `taf roles remove law test --keystore ..\keystore-dev`
- `taf targets add-repo law. --target-name test/test --role new-role --keystore ..\keystore-dev --serve latest` (new-role does't exist)
- `taf targets add-repo law. --target-name test/test2 --role new-role --keystore ..\keystore-dev --serve latest` (new-role exists and should be updated)
-` taf targets remove-repo law test/test --keystore ..\keystore-dev`
- `taf targets list law`

Tested majority of the old commands too, but makes sense to double-check

Closes #312
Closes #9 

## Code review checklist (for code reviewer to complete)

- [x] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR)
- [x] Title summarizes what is changing
- [x] Commit messages are meaningful (see [this][commit messages] for details)
- [ ] Tests have been included and/or updated, as appropriate
- [x] Docstrings have been included and/or updated, as appropriate
- [x] Changelog has been updated, as needed (see [CHANGELOG.md][changelog])

[changelog]: https://github.com/openlawlibrary/taf/blob/master/CHANGELOG.md
[commit messages]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Review by n-dusan — CHANGES_REQUESTED** [review]
Left some minor comments. I think we can merge after that. Not sure if it's within the scope of this PR, but could we update taf docs with examples on how to use newly added `taf roles` and `taf targets` commands? 

Do you think these examples can go in ## Setting up repositories section in https://github.com/openlawlibrary/taf/blob/master/docs/developers/testing_notes.md or is there another section that would be better?

Could we also update our https://github.com/openlawlibrary/taf/blob/master/docs/repo-creation-and-update-util.md and remove sections for commands that got removed in this PR? I think we need to update `update_and_sign_targets` documentation?

Since we now have `taf roles`, do we still need `taf metadata add-roles` command? What about `add-signing-key`?

File: taf/api/targets.py
Comment (renatav) [reply taf/api/targets.py]: I'm not sure, it might be more obvious that there is something that we might want to sign if keep `unsigned`. I don't feel strongly about it

File: taf/api/targets.py
Comment (renatav) [reply taf/api/targets.py:1]: I think that that is already how it works, just checked: [see]( https://github.com/openlawlibrary/taf/blob/b1e099d54516ad8b29a8137a1cd5f31aa1f3f851/taf/repositoriesdb.py#L75).

File: taf/tools/roles/__init__.py
Comment (renatav) [reply taf/tools/roles/__init__.py]: Done

File: docs/repo-creation-and-update-util.md
Comment (n-dusan) [inline docs/repo-creation-and-update-util.md:282]: In this case, which command should we be using instead?

File: docs/setting-up-new-library.md
Comment (n-dusan) [inline docs/setting-up-new-library.md:346]:
```suggestion
taf metadata update-expiration-date auth_repo_path targets --keystore keystore_path --interval days
```

File: taf/api/targets.py
Comment (n-dusan) [reply taf/api/targets.py:1]: Do we know what's causing slow output? Can we create an issue?

**Review by n-dusan — APPROVED** [review]
Approved, merge when ready

> @n-dusan All commands that are still in developer_tool are a bit different than the new commands. The plan is to improve them and mice from thar module so that we can eventually delete it.

Okay. After merging, could we also mention that plan in a new issue?

File: docs/repo-creation-and-update-util.md
Comment (renatav) [reply docs/repo-creation-and-update-util.md:282]: Well, we should use our build process that uses TAF, but is in another repo. So we could say something like, TAF can be used to implement an automated process which will update all repositories in accordance to a specific project's needs if that makes sense

**Comment by renatav** [comment]
@n-dusan I think that checking if a repo is up to date with the remote is the slowest operation, but we could profile. Maybe we could also add some output, like, checking if repositories are synced, that would make it more clear why it's taking a while.

File: docs/repo-creation-and-update-util.md
Comment (n-dusan) [reply docs/repo-creation-and-update-util.md:282]: Yes, that makes sense.

**Pull request merged**
